### PR TITLE
removed old migrations

### PR DESCRIPTION
**Pull request by debris**
- removed all migrations from 2016
- added migration to remove bloom group from database

### CODE REVIEW

File: util/migration/src/lib.rs [hunk 1]
```diff
@@ -138,7 +138,7 @@ pub trait SimpleMigration: 'static {
 	fn version(&self) -> u32;
 	/// Should migrate existing object to new database.
 	/// Returns `None` if the object does not exist in new version of database.
-	fn simple_migrate(&mut self, key: Vec<u8>, value: Vec<u8>) -> Option<(Vec<u8>, Vec<u8>)>;
+	fn simple_migrate(&mut self, key: Vec<u8>, value: Vec<u8>, col: Option<u32>) -> Option<(Vec<u8>, Vec<u8>)>;
```
Comment (debris) [inline util/migration/src/lib.rs]: This is the only new thing in this pr. I added it, cause it's good to know what column we are migrating

**Comment by 5chdn** [comment]
Does not compile

File: util/migration/src/lib.rs [hunk 2]
```diff
@@ -130,12 +130,14 @@ pub trait Migration: 'static {
...
 
-/// A simple migration over key-value pairs.
+/// A simple migration over key-value pairs of a single column.
 pub trait SimpleMigration: 'static {
 	/// Number of columns in database after the migration.
 	fn columns(&self) -> Option<u32>;
 	/// Version of database after the migration.
 	fn version(&self) -> u32;
+	/// Index of column which should be migrated.
+	fn migrated_column_index(&self) -> Option<u32>;
```
Comment (debris) [inline util/migration/src/lib.rs]: without this function, the trait is completely useless, cause we don't know which column we are currently migrating

**Comment by debris** [comment]
I think that migrations should be completely reworked. This pr makes a minimal effort take the most out of what we currently have.

**Comment by debris** [comment]
shoud be backported together with #7934

File: util/migration/src/lib.rs [hunk 3]
```diff
@@ -157,8 +160,12 @@ impl<T: SimpleMigration> Migration for T {
...
 
 		for (key, value) in iter {
-			if let Some((key, value)) = self.simple_migrate(key.into_vec(), value.into_vec()) {
-				batch.insert(key, value, dest)?;
+			if migration_needed {
+				if let Some((key, value)) = self.simple_migrate(key.into_vec(), value.into_vec()) {
+					batch.insert(key, value, dest)?;
+				}
+			} else {
+				batch.insert(key.into_vec(), value.into_vec(), dest)?;
```
Comment (tomusdrw) [inline util/migration/src/lib.rs]: By default we are just altering the database, do we really need to overwrite the data?

Comment (debris) [reply util/migration/src/lib.rs]: Unfortunately, we are not altering, but swapping :( 

https://github.com/paritytech/parity/blob/25b19835e316f53366df67b66244a8eee95bc865/util/migration/src/lib.rs#L291-L304

Comment (rphmeier) [reply util/migration/src/lib.rs]: yes, since it can fail we need to keep a backup until it completes

**Pull request merged**